suite-native/receive/src/components/DeviceScreenContent.tsx

```diff
@@ -89,18 +99,50 @@ export const DeviceScreenContent = ({
     isAddressRevealed,
     isPaginationEnabled,
     activePage,
+    isReceiveApproved,
+    isConfirmOnTrezorReady,
 }: DeviceScreenContentProps) => {
     const { applyStyle } = useNativeStyles();
     const deviceModel = useSelector(selectDeviceModel);
     const { fontSource, fontSize, lineWidth, lineHeight, pagerOffset } =
         deviceToContentStyles[deviceModel ?? 'T1B1'];
 
     const deviceFont = useFont(fontSource, fontSize);
+    const { translate } = useTranslate();
+    const [visibleDots, setVisibleDots] = useState<number>(1);
+    const [visualState, setVisualState] = useState<DeviceScreenVisualState>({
+        isAddressShown: false,
+        isPaginationShown: false,
+        isLoading: false,
+    });
+
+    useEffect(() => {
+        const isLoading = isAddressRevealed && !isConfirmOnTrezorReady && !isReceiveApproved;
+        const isAddressShown = isAddressRevealed && isConfirmOnTrezorReady;
```

Comment:
`isConfirmOnTrezorReady` already includes `isAddressRevealed`, same above. Please refactor these conditions to limit the amount of props. It's starting to become really messy and unpredictable. I aim to refactor this soon, but please simplify where possible

Reply:
rewritten